### DISCUSSION TITLE
Will ordering of multiple groupArray aggregations be consistent with each other?

### DISCUSSION THREAD

**Discussion by boginski** [ClickHouse/ClickHouse#42338] [Q&A] [answered]
If I select multiple groupArray, the same indexes in groupArray will contain data from the same row?

In other words:
"""
select
groupArray(some_field_1) as ga1,
groupArray(some_field_2) as ga2,
groupArray(some_field_3) as ga3
from table
"""

If I have a table with some_field_1, some_field_2, some_field_3. If ga1[i] contain data from k-th rows, then ga2[i] contain data from k-th rows? 

If it is possible, please send the link on the documentation.

**Comment by den-crane** [comment, marked as answer]
Yes, it will until you column contains Null (all aggregate functions including groupArray skip Nulls).
You can use Tuple to overcome it.

```sql
select
groupArray( (some_field_1, some_field_2, some_field_3) ) as ga
from table
```
or
```sql 
select
(groupArray( (some_field_1, some_field_2, some_field_3) )  as ga).1 ga1,
ga.2 ga2
ga.3 ga3
from table
```

**Reply by boginski** [reply]
Thank you very much ;)
If it was specified in the documentation, it would be wonderful)
